### PR TITLE
Replaces lag with lag(lite)

### PR DESCRIPTION
**Pull request by MajManatee**
# Document the changes in your pull request
Changes a fuckton of stuff to use delta_time instead of uhh whatever it was before.
idk jamie asked me to.

Ports https://github.com/tgstation/tgstation/pull/52981/files

:cl:   
tweak: delta_time is a thing that exists now
/:cl:

### CODE REVIEW

File: code/__DEFINES/atmospherics.dm [hunk 1]
```diff
@@ -200,9 +200,9 @@
 /// (kPa) What pressure pumps and powered equipment max out at.
 #define MAX_OUTPUT_PRESSURE					4500
 /// (L/s) Maximum speed powered equipment can work at.
-#define MAX_TRANSFER_RATE					200
-/// 10% of an overclocked volume pump leaks into the air
-#define VOLUME_PUMP_LEAK_AMOUNT				0.1
+#define MAX_TRANSFER_RATE					400
+/// How many percent of the contents that an overclocked volume pumps leak into the air
+#define VOLUME_PUMP_LEAK_AMOUNT				0.2
```
Comment (ToasterBiome) [inline code/__DEFINES/atmospherics.dm:205]: undocumented change

File: code/controllers/subsystem/fire_burning.dm [hunk 2]
```diff
@@ -32,7 +33,7 @@ SUBSYSTEM_DEF(fire_burning)
 
 		if(O.resistance_flags & ON_FIRE) //in case an object is extinguished while still in currentrun
 			if(!(O.resistance_flags & FIRE_PROOF))
-				O.take_damage(20, BURN, FIRE, 0)
+				O.take_damage(10 * delta_time, BURN, FIRE, 0)
```
Comment (ToasterBiome) [inline code/controllers/subsystem/fire_burning.dm:36]: undocumented change

File: code/controllers/subsystem/events.dm [hunk 3]
```diff
@@ -37,7 +37,7 @@ SUBSYSTEM_DEF(events)
 		var/datum/thing = currentrun[currentrun.len]
 		currentrun.len--
 		if(thing)
-			thing.process()
+			thing.process(wait * 0.1)
```
Comment (ToasterBiome) [inline code/controllers/subsystem/events.dm:40]: this is not how delta_time works

File: code/game/objects/items/candle.dm [hunk 4]
```diff
@@ -9,7 +9,7 @@
 	w_class = WEIGHT_CLASS_TINY
 	light_color = LIGHT_COLOR_FIRE
 	heat = 1000
-	var/wax = 1000
+	var/wax = 2000
```
Comment (ToasterBiome) [inline code/game/objects/items/candle.dm:12]: undocumented change

File: code/game/objects/items/candle.dm [hunk 5]
```diff
@@ -21,7 +21,7 @@
 		light()
 
 /obj/item/candle/update_icon()
-	icon_state = "candle[(wax > 400) ? ((wax > 750) ? 1 : 2) : 3][lit ? "_lit" : ""]"
```
Comment (ToasterBiome) [inline code/game/objects/items/candle.dm:24]: undocumented change

File: code/game/objects/items/chrono_eraser.dm [hunk 6]
```diff
@@ -167,7 +167,7 @@
 	interaction_flags_atom = NONE
 	var/mob/living/captured = null
 	var/obj/item/gun/energy/chrono_gun/gun = null
-	var/tickstokill = 15
+	var/timetokill = 30
```
Comment (ToasterBiome) [inline code/game/objects/items/chrono_eraser.dm:170]: undocumented change

File: code/game/objects/items/cigs_lighters.dm [hunk 7]
```diff
@@ -394,7 +394,7 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 	type_butt = /obj/item/cigbutt/cigarbutt
 	throw_speed = 0.5
 	item_state = "cigaroff"
-	smoketime = 300 // 11 minutes
+	smoketime = 11 * 60 // 11 minutes
```
Comment (ToasterBiome) [inline code/game/objects/items/cigs_lighters.dm:397]: use the defines

File: code/game/objects/items/cigs_lighters.dm [hunk 8]
```diff
@@ -349,7 +349,7 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 	type_butt = /obj/item/cigbutt/roach
 	throw_speed = 0.5
 	item_state = "spliffoff"
-	smoketime = 120 // four minutes
+	smoketime = 4 * 60
```
Comment (ToasterBiome) [inline code/game/objects/items/cigs_lighters.dm:352]: use the defines

**Review by ToasterBiome — CHANGES_REQUESTED** [review]
i dont think this is how a delta_time system is supposed to work

**Comment by Moltijoe** [comment]
Not sure if its the same, but during a brief period where i was coding something in unity delta time was basically the real time delay since last "tick"

**Comment by Hopekz** [comment]
the proper way to link a PR in another repo for credit is tgstation/tgstation#52981 vs pasting the link itself

**Comment by MajManatee** [comment]
jamie said copy the pr and add anything they changed so I did. take it up with him

**Comment by Hopekz** [comment]
> jamie said copy the pr and add anything they changed so I did. take it up with him

It's fine I already properly credited them for you so you don't have to do it since if you go on the link to the original PR you can see that this PR is now referenced there.

Using the proper format is just proper open source etiquette since if you put in the # format it "links" PR's together. 
Unrelated to Jamie.

**Comment by JamieD1** [comment]
> > jamie said copy the pr and add anything they changed so I did. take it up with him
> 
> It's fine I already properly credited them for you so you don't have to do it since if you go on the link to the original PR you can see that this PR is now referenced there.
> 
> Using the proper format is just proper open source etiquette since if you put in the # format it "links" PR's together. Unrelated to Jamie.

If we need it, not balance wise, Sorry miscommunication.

File: code/__DEFINES/atmospherics.dm (hunk 1 above)
Comment (JamieD1) [reply code/__DEFINES/atmospherics.dm:205]: Intended

File: code/controllers/subsystem/fire_burning.dm (hunk 2 above)
Comment (JamieD1) [reply code/controllers/subsystem/fire_burning.dm:36]: Is fine. Wait = 20 / 10 is 2 * 10 is 20.

File: code/game/objects/items/candle.dm (hunk 4 above)
Comment (JamieD1) [reply code/game/objects/items/candle.dm:12]: Intended

File: code/game/objects/items/candle.dm (hunk 5 above)
Comment (JamieD1) [reply code/game/objects/items/candle.dm:24]: intended

File: code/game/objects/items/chrono_eraser.dm (hunk 6 above)
Comment (JamieD1) [reply code/game/objects/items/chrono_eraser.dm:170]: intended

File: code/game/objects/items/cigs_lighters.dm (hunk 8 above)
Comment (JamieD1) [reply code/game/objects/items/cigs_lighters.dm:352]: its because its not seconds, its using deciseconds time the delta_time

**Pull request merged**